### PR TITLE
Add canSpaceJumpWaterEscape

### PR DESCRIPTION
**Pull request by kjbranch**

### CODE REVIEW

**Comment by kjbranch** [comment]
This addresses #700

File: region/crateria/east.json [hunk 1]
```diff
@@ -2452,6 +2455,29 @@
...
+              "id": 2,
+              "strats": [
+                {
+                  "name": "Space Jump Water Escape with Ice",
+                  "notable": false,
+                  "requires": [
+                    "canSpaceJumpWaterEscape",
+                    "canTrickyUseFrozenEnemies"
+                  ],
+                  "note": "Freeze the Choot and stand on it on the right side of the stalagmite. Jump to the water line and space jump to the Kamer platform."
```
Comment (osse101) [inline region/crateria/east.json:2468]: This has to be done in a very particular way.
Its a tricky jump to the right to get above the choot.
Break momentum to align with the wall.
And a small frame window shot timing on the freeze.

But attempts are repeatable without getting hit.
Could be made a notable.

**Review by osse101 — COMMENTED** [review]
There is a huge skill gap between `bouncing on water` and `turning the bounce into normal spacejump`.  So the tech needs more description.

File: region/maridia/inner-yellow.json [hunk 2]
```diff
@@ -857,7 +869,11 @@
                   "requires": [
                     "canSuitlessMaridia",
                     "HiJump",
-                    "canUseFrozenEnemies"
+                    "canUseFrozenEnemies",
+                    {"or": [
+                      "canWalljump",
+                      "h_canCrouchJumpDownGrab"
```
[the remark this answers is not in the document]
Comment (kjbranch) [reply region/maridia/inner-yellow.json]: Yes. but you need either a wall jump, crouch jump down grab, or space jump to escape the waterline

**Comment by kjbranch** [comment]
Thanks for the feedback. I think this was worth separating into two tech. I believe it is possible to get out of the water while space jumping without hjb, but it is extremely precise. I am pretty sure I got it in the Red Fish Room, but I can't replicate it - it may only be possible in tidal water.

File: region/crateria/east.json [hunk 3]
```diff
@@ -2408,7 +2408,10 @@
                   "requires": [
                     "canSuitlessMaridia",
                     "SpaceJump",
```
Comment (osse101) [inline region/crateria/east.json:2410]: spacejumpWaterBounce?

File: region/maridia/outer.json [hunk 4]
```diff
@@ -4376,6 +4376,25 @@
                     ]}
                   ]
                 },
+                {
+                  "name": "Suitless Space Jump with Wall Jump",
```
Comment (osse101) [inline region/maridia/outer.json:4380]: Rename

File: region/maridia/inner-yellow.json [hunk 5]
```diff
@@ -872,7 +872,8 @@
                     "canUseFrozenEnemies",
                     {"or": [
                       "canWalljump",
-                      "h_canCrouchJumpDownGrab"
+                      "h_canCrouchJumpDownGrab",
```
Comment (osse101) [inline region/maridia/inner-yellow.json:875]: I still can't get the jump off the fish without `CrouchJumpDownGrab`

**Review by osse101 — COMMENTED** [review]
I like the split into 2 techs.

`canSpaceJumpWaterEscape` makes sense as a tech but its too bad that in practice its just a much harder way of avoiding `canWalljump`

File: region/maridia/inner-yellow.json (hunk 5 above)
Comment (kjbranch) [reply region/maridia/inner-yellow.json]: Whoops. You're right. I was confusing myself

File: tech.json [hunk 6]
```diff
@@ -51,6 +51,21 @@
...
+              "requires": [
+                "canSuitlessMaridia",
+                "SpaceJump"
+              ],
+              "note": "Using Space Jump to bounce along the water line. It can be used to escape the water with a wall jump, or to get onto low platforms just above the water level.",
+              "extensionTechs": [
+                {
+                  "name": "canSpaceJumpWaterEscape",
+                  "requires": [ "canSpaceJumpWaterBounce" ],
+                  "note": "Using Space Jump with precise timing at the water line to escape water without Gravity Suit."
```
Comment (osse101) [inline tech.json:66]: I'd like a mention of `Space Jump Water Bounce` here.
Converting Space Jump Water Bouncing into regular space jumps without Gravity.  Often paired with HiJump.

**Pull request merged**